packages/web-components/@sfx/search-box/src/search-box.ts

```diff
@@ -0,0 +1,205 @@
+    this.dispatchEvent(
+      new CustomEvent(SEARCHBOX_EVENT.SEARCHBOX_CHANGE, {
+        detail: (e.target as HTMLInputElement).value,
+        bubbles: true
+      })
+    );
+  }
+
+  /**
+   * Invokes the emitSearchEvent function when a user presses enter and 
+   * the value property length is greater than 0.
+   *
+   * @param e The KeyboardEvent object.
+   */
+  handleKeydown(e: KeyboardEvent) {
+    if (e.keyCode === KEY_CODES.ENTER && this.value.length > 0) {
+      this.emitSearchEvent();
+    }
+  }
+
+  /**
+   * Updates the value property to the value passed to it.
+   *
+   * @param term The value pulled directly from the input box.
+   */
+  updateSearchTermValue(term: string) {
+    this.value = term;
+  }
+
+  /**
+   * Sets value property and searchbox input value to an empty string and
+   * invokes the emitSearchBoxClearedEvent function.
+   * Invoked in response to click on `clear` button.
+   */
+  clearSearch() {
+    this.getInputElement().value = '';
+    this.value = '';
+    this.emitSearchBoxClearClick();
+  }
+
+  /**
+   * Dispatches an event notifying that the search box input bar has been clicked.
+   * Invoked in response to a user clicking inside of the searchbox input.
+   */
+  clickExposed() {
+    const searchBoxClickedEvent = new CustomEvent(SEARCHBOX_EVENT.SEARCHBOX_CLICK, { bubbles: true });
+    this.dispatchEvent(searchBoxClickedEvent);
+  }
+
+  /**
+   * Dispatches an event notifying that the search box input bar has been hovered.
+   * Invoked in response to a user hovering inside of the searchbox input.
+   */
+  hoverExposed() {
+    const searchBoxHoveredEvent = new CustomEvent(SEARCHBOX_EVENT.SEARCHBOX_HOVER, { bubbles: true });
+    this.dispatchEvent(searchBoxHoveredEvent);
+  }
+
+  // TODO Move this to the Storybook tab once functionality has been merged into sfx-view.
+  /*
+   * --- TEMPORARY: setup for testing event listeners ---
+   * Because Storybook is contained in an iframe, for testing purposes,
+   * we are unable to dispatch events directly from the console.
+   *
+   * As an alternative - temporarily, we have a button to click to dispatch event.
+   * Should be updated when/if functionality is avaiable via Storybook tab.
+   * This event would be dispatched when a user hovers on autocomplete.
+   */
+  updateTextEvent() {
+    const updateSearchTerm = new CustomEvent(SEARCHBOX_EVENT.UPDATE_SEARCH_TERM, {
+      detail: 'catfood',
+      bubbles: true
+    });
+    window.dispatchEvent(updateSearchTerm);
+  }
+
+  render() {
+    return html`
+    <input 
+      type="text" 
+      @mouseenter="${this.hoverExposed}" 
+      @click="${this.clickExposed}" 
+      data-sfx-ref="searchInput" 
+      placeholder="${this.placeholder}" 
+      @input="${this.handleChange}"
+      @keydown="${this.handleKeydown}">
+    </input>
+    ${
+      this.clearButton
+        ? html`<button @click="${this.clearSearch}">Clear</button>`
```

Comment:
No action needed for now, but do we want to consider setting the `type` of this button to `reset` in case the client puts this inside a `<form>`? If so, `clearSearch` would need to prevent default on the event.